resources/devbox/macos/macos.sh

```diff
@@ -0,0 +1,200 @@
+
+set -e
+set -u
+
+## Note: does not use standard build script include
+# adjust relative paths as necessary. Note that this will not work in a symlinked path reliably
+THIS_SCRIPT="${BASH_SOURCE[0]}"
+THIS_DIR=$(dirname "$THIS_SCRIPT")
+
+# Determine which components to install
+REQUIRE_ANDROID=false
+REQUIRE_IOS=false
+REQUIRE_MACOS=false
+REQUIRE_WEB=false
+
+# Optional components
+REQUIRE_KMCOMP=false
+REQUIRE_PANDOC=false
+REQUIRE_SENTRYCLI=false
+
+# Parse args
+shopt -s nocasematch
+
+PARAMFOUND=false
+
+function print_help() {
+    echo "Usage: macos.sh targets"
+    echo "  targets: one or more of: "
+    echo "             android ios macos web all"
+    echo "  optional targets:"
+    echo "             kmcomp pandoc sentry-cli"
+    echo "Targets will automatically include dependency targets."
+}
+
+while [[ $# -gt 0 ]] ; do
+    key="$1"
+    case $key in
+        --help)
+            print_help
+            exit 0
+            ;;
+        android)
+            REQUIRE_ANDROID=true
+            PARAMFOUND=true
+            ;;
+        ios)
+            REQUIRE_IOS=true
+            PARAMFOUND=true
+            ;;
+        macos)
+            REQUIRE_MACOS=true
+            PARAMFOUND=true
+            ;;
+        web)
+            REQUIRE_WEB=true
+            PARAMFOUND=true
+            ;;
+        kmcomp)
+            REQUIRE_KMCOMP=true
+            PARAMFOUND=true
+            ;;
+        pandoc)
+            REQUIRE_PANDOC=true
+            PARAMFOUND=true
+            ;;
+        sentry-cli)
+            REQUIRE_SENTRYCLI=true
+            PARAMFOUND=true
+            ;;
+        all)
+            REQUIRE_ANDROID=true
+            REQUIRE_IOS=true
+            REQUIRE_MACOS=true
+            REQUIRE_WEB=true
+            PARAMFOUND=true
+            ;;
+        *)
+            echo "Error: unrecognised parameter."
+            echo
+            print_help
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+if ! $PARAMFOUND; then
+    echo "Error: must have a target parameter"
+    echo
+    print_help
+    exit 1
+fi
+
+# This script will configure your development environment from a bare metal mac install. It should be idempotent.
+
+echo "This script will configure your macOS computer to build Keyman, installing build tools and prerequisites."
+echo "You can also do this yourself following the notes in building.md."
+echo
+read -p "Press ENTER to start install"
+
+if $REQUIRE_IOS || $REQUIRE_ANDROID; then
+  REQUIRE_WEB=true
+fi
+
+if $REQUIRE_ANDROID; then
+  REQUIRE_PANDOC=true
+fi
+
+## These components are required by all development toolchains
+
+# XCode command lines tools will be installed by Brew
+# xcode-select --install
+
+which brew || (
+    # Install Homebrew
+    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+) && (
+    brew update
+)
+
+## Install devchain components
+
+BREW_ALL="bash jq python3 meson ninja rustup-init coreutils"
```

Comment:
Regarding my question in this comment chain: https://github.com/keymanapp/keyman/pull/5324/files#r655827748

If `meson` and `ninja` are solely for `common/core/desktop`, why are we including it for Web & Web-based builds?

Reply:
Yes, this is prep for the integration for LDML.